datafusion/expr/src/planner.rs

```diff
@@ -60,6 +60,9 @@ pub trait ContextProvider {
         not_impl_err!("Recursive CTE is not implemented")
     }
 
+    /// Getter for expr planners
+    fn get_expr_planners(&self) -> Vec<Arc<dyn ExprPlanner>>;
```

Comment:
I spent some time trying to figure out if we could avoid this Vec cloning all the time -- with

```suggestion
    fn get_expr_planners(&self) -> &[Arc<dyn ExprPlanner>] {&[]}
```

I think this is possible -- I made https://github.com/jayzhan211/arrow-datafusion/pull/3 to show how it looks